addons/sms/wizard/sms_composer.py

```diff
@@ -210,7 +210,7 @@ def _action_send_sms_comment_single(self, records=None):
         # on the numbers in the database.
         records = records if records is not None else self._get_records()
         records.ensure_one()
-        if not self.number_field_name:
+        if not self.number_field_name or self.number_field_name not in records:
```

Comment:
We can keep the check since we already have fallbacks anyway but `Stock Picking` defines `_sms_get_number_fields`, which says it should have a mobile and/or phone field to pick from. So there's an issue there too.

It should defined `_sms_get_partner_fields` instead so we can fallback on the partner instead. Such that this condition would not be reached.